Seems like ColorButton is not working (Windows 10)

Wanted to check `ColorButton` component but instead got error. Clean project created from starter.

Dependencies:

```
"dependencies": {
    "proton-native": "^1.1.1"
  },
  "devDependencies": {
    "babel-cli": "^6.26.0",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "electron-builder": "^20.11.1"
  }
```

Whole application code:

```
import React, { Component } from "react";
import { render, Window, App, ColorButton } from "proton-native";

class Example extends Component {
  render() {
    return (
      <App>
        <Window
          title="Proton Native Rocks!"
          size={{ w: 300, h: 300 }}
          menuBar={false}
        >
          <ColorButton />
        </Window>
      </App>
    );
  }
}

render(<Example />);

```

System: `Win 10 17134 (2018 April Update)`, Node: `9.3`

Error:

```
The above error occurred in the <COLORBUTTON> component:
    in COLORBUTTON (created by Example)
    in WINDOW (created by Example)
    in APP (created by Example)
    in Example

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
C:\bin\projects\workspace\jurosh\experimental\experiments\proton-react-native\node_modules\react-reconciler\cjs\react-reconciler.development.js:6780
      throw _error4;
      ^

TypeError: Type mismatch
    at _loop2 (C:\bin\projects\workspace\jurosh\experimental\experiments\proton-react-native\node_modules\proton-native\bin\components\ColorButton.js:123:32)
    at ColorButton.initialProps (C:\bin\projects\workspace\jurosh\experimental\experiments\proton-react-native\node_modules\proton-native\bin\components\ColorButton.js:129:9)
    at new ColorButton (C:\bin\projects\workspace\jurosh\experimental\experiments\proton-react-native\node_modules\proton-native\bin\components\ColorButton.js:49:11)
    at Object.COLORBUTTON (C:\bin\projects\workspace\jurosh\experimental\experiments\proton-react-native\node_modules\proton-native\bin\utils\createElement.js:50:14)
    at createElement (C:\bin\projects\workspace\jurosh\experimental\experiments\proton-react-native\node_modules\proton-native\bin\utils\createElement.js:133:26)
    at createInstance (C:\bin\projects\workspace\jurosh\experimental\experiments\proton-react-native\node_modules\proton-native\bin\reconciler\index.js:26:45)
    at completeWork (C:\bin\projects\workspace\jurosh\experimental\experiments\proton-react-native\node_modules\react-reconciler\cjs\react-reconciler.development.js:4427:31)
    at completeUnitOfWork (C:\bin\projects\workspace\jurosh\experimental\experiments\proton-react-native\node_modules\react-reconciler\cjs\react-reconciler.development.js:5918:18)
    at performUnitOfWork (C:\bin\projects\workspace\jurosh\experimental\experiments\proton-react-native\node_modules\react-reconciler\cjs\react-reconciler.development.js:6020:14)
    at workLoop (C:\bin\projects\workspace\jurosh\experimental\experiments\proton-react-native\node_modules\react-reconciler\cjs\react-reconciler.development.js:6074:26)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! proton-react-native@0.0.1 start: `babel-node index.js`
```

--- a/src/components/ColorButton.js
+++ b/src/components/ColorButton.js
@@ -6,6 +6,10 @@ import libui from 'libui-node';
 import Color from 'color';
 import PropTypes from 'prop-types';
 
+const functionMappings = {
+  onChange: 'onChanged',
+};
+
 class ColorButton extends DesktopComponent {
   eventParameter = { onChanged: 'color' };
 
@@ -41,13 +45,19 @@ class ColorButton extends DesktopComponent {
       // normal props
       if (oldProps[prop] !== newProps[prop] && prop !== 'color') {
         // add check for color prop
-        if (typeof props[prop] === 'function') {
-          if (this.eventParameter[prop] !== '') {
-            this.element[prop](() =>
-              newProps[prop](this.element[this.eventParameter[prop]])
+        if (typeof newProps[prop] === 'function') {
+          const translatedProp = functionMappings[prop]; // translate React function names in libui names
+          if (typeof this.eventParameter[translatedProp] === 'function') {
+            // if we don't have a translatedProperty, then we use a function, so handle that
+            this.element[translatedProp](() =>
+              newProps[prop](this.eventParameter[translatedProp]())
+            );
+          } else if (this.eventParameter[translatedProp] !== '') {
+            this.element[translatedProp](() =>
+              newProps[prop](this.element[this.eventParameter[translatedProp]])
             );
           } else {
-            this.element[prop](newProps[prop]);
+            this.element[translatedProp](newProps[prop]);
           }
         } else if (prop == 'children') {
           this.element[childName] = newProps[prop];
@@ -66,17 +76,23 @@ class ColorButton extends DesktopComponent {
     for (let prop in props) {
       // normal props
       if (typeof props[prop] === 'function') {
-        if (this.eventParameter[prop] !== '') {
-          this.element[prop](() =>
-            props[prop](
-              this.toRgbObject(this.element[this.eventParameter[prop]])
-            )
+        const translatedProp = functionMappings[prop]; // translate React function names in libui names
+        if (typeof this.eventParameter[translatedProp] === 'function') {
+          // if we don't have a property, then we use a function, so handle that
+          this.element[translatedProp](() =>
+            props[prop](this.eventParameter[translatedProp]())
+          );
+        } else if (this.eventParameter[translatedProp] !== '') {
+          this.element[translatedProp](() =>
+            props[prop](this.element[this.eventParameter[translatedProp]])
           );
         } else {
-          this.element[prop](props[prop]);
+          this.element[translatedProp](props[prop]);
         }
       } else if (prop == 'children') {
         this.element[this.childName] = props[prop];
+      } else if (prop === 'color') {
+        this.element[prop] = this.convertToColor(props[prop]);
       } else {
         this.element[prop] = props[prop];
       }

